src/orderbook/state.ts

```diff
@@ -146,6 +252,47 @@ export class AccountState {
     return account
   }
 
+  /**
+   * Updates a user's account balance.
+   *
+   * @throws If the balance underflows, unless the `allowUnderflow` option is
+   * specified.
```

Comment:
Doesn't actually throw (and shouldn't because it can be temporarily negative within one batch).

Reply:
It used to, but the check moved to assertAccountsAreValid because of the reason you mentioned. I'll update the comment.